### PR TITLE
Forces a confirmation page on 'Forgot Password' page even if user does not exist

### LINKED ISSUE #3639: user/forget_password confirmation/error messages should be changed
You should change the error output the page "user/forget_password" gives when the email address is not known. Currently it is "This email address is not associated with any account." while it should just print a positive confirmation like mailman does[0] something like "if this email account exists (etc etc)", likewise the positive result "A new confirmation email has been sent to name@host.tld, please check your inbox within the next 3 hours to complete the registration process." should be something similar.

At the moment I don't know enough about gogs to fix it myself, I'm just packaging it.

[0]: "If you are a list member, your password has been emailed to you."

### PR DESCRIPTION
**Pull request by LefsFlarey**
# TLDR; Closes #3639 
This small hot-fix is basically a very cheat way of showing a successful 'confirmation email' sent to the email even if it does not exists.

# Why?
Well, by allowing users to see if an email exists breaks a very fundamental law of security through obscurity. This small patch just forces Gogs to show a confirmation page even if the user does not exist.

### CODE REVIEW

**Comment by LefsFlarey** [comment]
Ah, I'm not sure if @Unknwon actually wants this feature. In my opinion, it's important that validity of email address should not be disclosed and this patch ensures confidentiality of your user's emails.

**Comment by Unknwon** [comment]
Thanks!
